src/frequenz/channels/util/_file_watcher.py

```diff
@@ -14,7 +14,7 @@
 from .._exceptions import ReceiverStoppedError
 
 
-class FileWatcher(Receiver[pathlib.Path]):
+class FileWatcher(Receiver[Tuple[Change, pathlib.Path]]):
```

Comment:
This should actually return a `FileWatcher.EventType`, the underlying library used to implement `FileWatcher` is hidden to the user.

I would also recommended using a `dataclass` as the return:

```python
@dataclass(frozen=True)
class Event:
    type: EventType
    path: pathlib.Path
```

It avoids errors and makes the code more readable and future proof (if we add more attributes users don't need to update the code (if they are using tuple unpacking syntax, adding a new item to the tuple will make `(change, path) = fw.receive()` fail).

Reply:
this suggestion makes totally sense, I'll update it accordingly. Thanks!